pkg/build/controller/build/build_controller.go

```diff
@@ -364,14 +375,24 @@ func (bc *BuildController) createPodSpec(originalBuild *buildapi.Build, ref stri
 	return podSpec, nil
 }
 
-// resolveOutputDockerImageReference returns a reference to a Docker image
-// computed from the buid.Spec.Output.To reference.
-func (bc *BuildController) resolveOutputDockerImageReference(build *buildapi.Build) (string, error) {
+// resolveOutputDockerImageReference returns a Docker image name
+// computed from the build.Spec.Output.To reference.  It also attempts to find a Secret
+// associated with the build service account which can push to the image location and
+// return that, if found.
+// Note that we are using controller level permissions to resolve the imagestream and secret,
+// meaning users could theoretically define a build that references an imagestream they cannot
+// see, and 1) get the docker image reference of that imagestream and 2) a reference to a secret
+// associated with a service account that can push to that location.  However they still cannot view the secret,
```

Comment:
The builder pod never gets access to the push secret?  I expected that you'd be passing that into the container doing the build so that it could push its output.  Doing so means that the secret is exposed to the end user.